From e3fdad047aee848f30c3170cb17f569b0342513d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Klaus=20K=C3=A4mpf?= <kkaempf@suse.de>
Date: Tue, 28 Nov 2017 15:37:19 +0100
Subject: [PATCH 22/25] Fix crash caused by NULL content_type

Author: Adam Majer <adam.majer@suse.de>

Also, allow requests with Content-Type set to text/xml
---
 cimXmlParser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: sblim-sfcb-1.4.9/cimXmlParser.c
===================================================================
--- sblim-sfcb-1.4.9.orig/cimXmlParser.c
+++ sblim-sfcb-1.4.9/cimXmlParser.c
@@ -1738,7 +1738,7 @@ scanCimXmlRequest(CimRequestContext *ctx
   control.MQs = 0;
   control.MPQs = 0;
 
-  if (strncmp(ctx->contentType,"application/xml",15) !=0 ) {
+  if (strncmp(ctx->contentType,"application/xml",15) !=0 && strncmp(ctx->contentType,"text/xml",8) != 0) {
     *rc=1;
     return control.reqHdr;
   }
Index: sblim-sfcb-1.4.9/httpAdapter.c
===================================================================
--- sblim-sfcb-1.4.9.orig/httpAdapter.c
+++ sblim-sfcb-1.4.9/httpAdapter.c
@@ -1048,7 +1048,7 @@ doHttpRequest(CommHndl conn_fd)
 
   inBuf.authorization = "";
   inBuf.protocol = "HTTP/1.1";
-  inBuf.content_type = NULL;
+  inBuf.content_type = "application/xml";
   inBuf.content_length = UINT_MAX;
   inBuf.host = NULL;
   inBuf.useragent = "";
